lib/bolt/executor.rb

```diff
@@ -26,39 +34,22 @@ def initialize(config = Bolt::Config.new, noop = nil, plan_logging = false)
       @notifier = Bolt::Notifier.new
     end
 
-    def from_targets(targets)
-      targets.map do |target|
-        Bolt::Node.from_target(target)
-      end
-    end
-    private :from_targets
-
-    def on(nodes, callback = nil)
+    def on(targets, callback = nil)
       results = Concurrent::Array.new
 
-      poolsize = [nodes.length, @config[:concurrency]].min
+      poolsize = [targets.length, @config[:concurrency]].min
       pool = Concurrent::FixedThreadPool.new(poolsize)
       @logger.debug { "Started with #{poolsize} thread(s)" }
 
-      nodes.map(&:class).uniq.each do |klass|
-        klass.initialize_transport(@logger)
-      end
-
-      nodes.each { |node|
+      targets.each { |target|
         pool.post do
           result =
             begin
-              @notifier.notify(callback, type: :node_start, target: node.target) if callback
-              node.connect
-              yield node
+              transport = @transports[target.protocol || 'ssh']
```

Comment:
I don't think we need this when running. We had to handle this default in inventory or we couldn't have loaded config options properly.

Reply:
I guess if a user set nil as transport in config?

Reply:
This was really just here for tests, so we don't have to put a protocol on all the URIs or call `update_conf` every time.